src/dot.ts

```diff
@@ -35,20 +32,7 @@ export class Dot extends Modifier {
       const dot = dots[i];
       const note = dot.getNote();
 
-      let props;
-      let shift;
-
-      if (isStaveNote(note)) {
-        const index = dot.checkIndex();
-        props = note.getKeyProps()[index];
-        shift = note.getRightDisplacedHeadPx();
-      } else if (isTabNote(note)) {
-        props = { line: 0.5 }; // Shim key props for dot placement
-        shift = 0;
-      } else {
-        // note object is not StaveNote or TabNote.
-        throw new RuntimeError('Internal', 'Unexpected instance.');
-      }
+      const { props, shift } = note.getDotPropsAndShift(dot.checkIndex());
```

Comment:
The previous block of code is replaced with this single line. Instead of Dot checking the note instance's type, we just let the note instance do its own thing. StaveNote and TabNote implement this method differently. This removes the dependencies.